tests/rptest/tests/under_replicated_metric_test.py

```diff
@@ -0,0 +1,161 @@
+
+import sys
+
+
+class UnderReplicatedMetricTest(EndToEndTest):
+    def __init__(self, test_context):
+        extra_rp_conf = dict(enable_leader_balancer=False,
+                             default_topic_replications=3,
+                             group_topic_partitions=3)
+
+        super(UnderReplicatedMetricTest,
+              self).__init__(test_context=test_context,
+                             extra_rp_conf=extra_rp_conf)
+
+    def check_no_under_replicated(self, sampler):
+        for n, points in sampler._samples.items():
+            for s in points[-1].sample.samples:
+                self.logger.info(f"ms: {s}")
+                assert s.value == 0
+
+    @cluster(num_nodes=6)
+    def test_under_replicated_metric_no_failures(self):
+        self.start_redpanda(num_nodes=3)
+        # create topics
+        topics = []
+        for _ in range(0, 6):
+            topics.append(TopicSpec(partition_count=random.randint(1, 10)))
+        # chose one topic to run the main workload
+        DefaultClient(self.redpanda).create_topic(topics)
+        self.topic = random.choice(topics).name
+
+        self.start_producer(1)
+        self.start_consumer(2)
+        self.await_startup()
+
+        sampler = MetricsSampler(self.test_context,
+                                 self.redpanda,
+                                 metrics_pattern=['.*under_replicated.*'])
+        sampler.start()
+
+        self.run_validation(min_records=20000,
+                            enable_idempotence=False,
+                            producer_timeout_sec=60,
+                            consumer_timeout_sec=180)
+        sampler.stop()
+        # all partitions should be fully replicated
+        self.check_no_under_replicated(sampler)
+
+    @cluster(num_nodes=4)
+    @parametrize(acks=1)
+    @parametrize(acks=-1)
+    def test_under_replicated_metric_with_different_acks(self, acks):
+        self.start_redpanda(num_nodes=3)
+        # create topics
+        topics = []
+        for _ in range(0, 6):
+            topics.append(TopicSpec(partition_count=random.randint(1, 10)))
+        # chose one topic to run the main workload
+        DefaultClient(self.redpanda).create_topic(topics)
+        self.topic = random.choice(topics).name
+
+        rpk_producer = RpkProducer(self.test_context,
+                                   self.redpanda,
+                                   self.topic,
+                                   16384,
+                                   sys.maxsize,
+                                   acks=acks)
+
+        sampler = MetricsSampler(self.test_context,
+                                 self.redpanda,
+                                 interval_seconds=2,
+                                 metrics_pattern=['.*under_replicated.*'])
+        sampler.start()
+        rpk_producer.start()
+        # sleep to collect metrics samples
+        sleep(16)
+        sampler.stop()
+        self.check_no_under_replicated(sampler)
```

Comment:
This sort of looks like the existing `MetricsCheck` will do this. From `rptest/services/metrics_check.py`:

```
A MetricCheck spans a region of code: instantiate at the start, then
call `expect` or `evaluate` later to measure how your metrics of interest
have changed over that region.
```

Reply:
oh, i created this PR earlier, i think MetricCheck will be enough